### ISSUE TITLE
Amplify Storage does not use credentials from Auth

### ISSUE THREAD

**Issue by mllrmat** [aws-amplify/amplify-js#3008]
**Describe the bug**
Amplify Storage module does not use the credentials from Cognito to upload files, but the local aws credentials found on the machine. 

**To Reproduce**
1. configure aws credentials for use on the command line
2. setup a nodejs app using cognito as authentication
3. sign in with cognito
4. upload a file with the nodejs client

**Expected behavior**
Amplify uses the credentials configured and where the rest of amplify is authenticated

**Desktop (please complete the following information):**
 - OS: osx, nodejs 10.12.0

**Comment by elorzafe** [comment]
@mllrmat amplify is officially not supported on nodejs. The library is focused to help on client apps web/mobile. 

Can you paste your app code so I can try to reproduce it locally.

Thanks for your feedback!

**Comment by komushi** [comment]
@elorzafe @mllrmat 

I have the similar issue. In the result of Auth.currentAuthenticatedUser() the "storage" attribute is missing.

Below is my node.js code:
```
const aws_exports = require('./aws-exports').getConfig();
const Amplify = require('aws-amplify');
global.fetch = require('node-fetch');
global.navigator = {};


Amplify.default.configure({
    Auth: {
        identityPoolId: aws_exports.aws_cognito_identity_pool_id,
        region: aws_exports.aws_cognito_region,
        userPoolId: aws_exports.aws_user_pools_id,
        userPoolWebClientId: aws_exports.aws_user_pools_web_client_id,
    },
    Storage: {
        AWSS3: {
            bucket: aws_exports.aws_user_files_s3_bucket,
            region: aws_exports.aws_user_files_s3_bucket_region
        }
    }
});

Amplify.Storage.configure({ level: 'protected' });

const username = 'user';
const password = 'password';


const proceedAuth = async () => {
	const Auth = Amplify.Auth;
	const rtn = await Auth.signIn(username, password).catch((err) => {

	  throw err;
	});

	return rtn;
}

const loadAuthenticationInfo = async() => {

	const Auth = Amplify.Auth;

	const [
	  currentAuthenticatedUser, 
	  currentUserInfo, 
	] = 
	  await Promise.all([
	    Auth.currentAuthenticatedUser(), 
	    Auth.currentUserInfo(),
	  ]);

	 // The issue is inside the currentAuthenticatedUser object,
	 // the "storage" attribute under root is missing.
	console.log('currentAuthenticatedUser', JSON.stringify(currentAuthenticatedUser));

	return { fileKey: Math.floor(Date.now() / 1000), data: currentUserInfo }

}

const saveFile = async ({fileKey, data}, fileContentType, s3level) => {

    await Amplify.Storage.put(
    	fileKey, 
    	JSON.stringify(data), { 
    		level: s3level,
			contentType: 'application/json'
		}).catch(e => {
      console.log("storage.put", e);
      throw e;
    });

}

proceedAuth()
	.then(loadAuthenticationInfo)
	.then(saveFile)
	.catch(e => console.error(e));
```

Below is the result of Auth.currentAuthenticatedUser():
```
{
  "username": "xu",
  "pool": {
  	"contents omitted": "contents omitted"
  },
  "Session": null,
  "client": {
  	"contents omitted": "contents omitted"
  },
  "signInUserSession": {
    "idToken": {
      "contents omitted": "contents omitted"
    },
    "refreshToken": {
      "contents omitted": "contents omitted"
    },
    "accessToken": {
      "contents omitted": "contents omitted"
    },
    "clockDrift": 0
  },
  "authenticationFlowType": "USER_SRP_AUTH",
  "keyPrefix": "contents omitted",
  "userDataKey": "contents omitted",
  "attributes": {
    "contents omitted": "contents omitted"
  },
  "preferredMFA": "NOMFA",
  "storage": "<= the missing attibute in node.js mode but available in browser"
}
```

**Comment by mbudm** [comment]
I ran into this issue too - I'm using Amplify in my node.js functional test to try and closely replicate what the web client does. 

I was looking into changing over to use the [aws4](https://github.com/mhart/aws4/issues/69) library to do my signed S3 uploads and stumbled on an idea. Why not simply forcibly set the AWS.config?

It seems to work. Once I've configured Amplify (Auth and Storage) and signed in with my test Cognito user, I use `Amplify.Auth.currentUserCredentials()` to get the credentials and assign it to AWS.config.

I've got a small module in my test setup that was collecting the Cognito IdentityId. This was as good a spot as any to force update the AWS credentials.

```
import Amplify from "aws-amplify";
import * as AWS from "aws-sdk";

export function getIdentityId(response: any) {
  return Amplify.Auth.currentUserCredentials()
    .then((res: any) => {
      AWS.config.credentials = res; // assing the cognito credentials to aws-sdk
      return {
        userIdentityId: res.params.IdentityId,
        ...response,
      };
    });
}
```
